mixer/adapter/solarwinds/metrics_handler.go

```diff
@@ -89,34 +90,32 @@ func newMetricsHandler(ctx context.Context, env adapter.Env, cfg *config.Params)
 		lc:          lc,
 		persistWait: persistWait,
 		batchWait:   batchWait,
+		metricInfo:  cfg.Metrics,
 	}, nil
 }
 
 func (h *metricsHandler) handleMetric(_ context.Context, vals []*metric.Instance) error {
 	measurements := []*appoptics.Measurement{}
 	for _, val := range vals {
-		merticVal := h.aoVal(val.Value)
-
-		m := &appoptics.Measurement{
-			Name:  val.Name,
-			Value: merticVal,
-			Time:  time.Now().Unix(),
-			Tags:  appoptics.MeasurementTags{},
-		}
+		if mInfo, ok := h.metricInfo[val.Name]; ok {
+			merticVal := h.aoVal(val.Value)
+
+			m := &appoptics.Measurement{
+				Name:  val.Name,
+				Value: merticVal,
+				Time:  time.Now().Unix(),
+				Tags:  appoptics.MeasurementTags{},
+			}
 
-		for k, v := range val.Dimensions {
-			switch vv := v.(type) {
-			case int:
-			case int32:
-			case int64:
-				m.Tags[k] = strconv.FormatInt(vv, 10)
-			case float64:
-				m.Tags[k] = strconv.FormatFloat(vv, 'f', -1, 64)
-			default:
-				m.Tags[k], _ = v.(string)
+			for k, v := range val.Dimensions {
```

Comment:
Considering labels is an array and Dimensions is a map, and we have already validated labels, do you think we can replace these loops with:
```
for _, label := range mInfo.LabelNames {
// val.Dimensions[label] should exists because we have validated this before during config time.
m.Tags[label] = h.processLabels(val.Dimensions[label]) 
}
```
I think it might work.. WDYT ?